c10/core/SymInt.cpp

```diff
@@ -1,95 +1,94 @@
 #include <c10/core/SymFloat.h>
 #include <c10/core/SymInt.h>
-#include <c10/core/SymIntNodeImpl.h>
+#include <c10/core/SymNodeImpl.h>
 #include <array>
 
 namespace c10 {
 
-static std::array<SymIntNode, 2> normalize_symints(SymInt a_, SymInt b_) {
-  SymIntNode a, b;
+static std::array<SymNode, 2> normalize_symints(SymInt a_, SymInt b_) {
+  SymNode a, b;
   if (a_.is_symbolic())
-    a = a_.toSymIntNodeImpl();
+    a = a_.toSymNodeImpl();
   if (b_.is_symbolic())
-    b = b_.toSymIntNodeImpl();
+    b = b_.toSymNodeImpl();
 
-  SymIntNodeImpl* common = a ? a.get() : b.get();
+  SymNodeImpl* common = a ? a.get() : b.get();
   // TODO: technically we need to check that the classes match
   if (!a) {
-    a = common->wrap(a_.as_int_unchecked());
-    a_.toSymInt(a); //
+    a = common->wrap_int(a_.as_int_unchecked());
   }
   if (!b) {
-    b = common->wrap(b_.as_int_unchecked());
-    b_.toSymInt(b);
+    b = common->wrap_int(b_.as_int_unchecked());
   }
   return {a, b};
 }
 
-SymIntNode SymInt::toSymIntNodeImpl() const {
+SymNode SymInt::toSymNodeImpl() const {
   TORCH_CHECK(is_symbolic());
-  return SymIntNode::reclaim_copy(toSymIntNodeImplUnowned());
+  return SymNode::reclaim_copy(toSymNodeImplUnowned());
 }
 
-c10::SymInt SymInt::toSymInt(SymIntNode sin_sp) {
+SymInt::SymInt(SymNode sin_sp) {
+  TORCH_CHECK(sin_sp->is_int());
```

Comment:
should this be an internal assert? or can user code end up doing this

Reply:
I think this shouldn't be possible to exercise from Python, but you can definitely trigger this relatively easily in c++. No big pref from me.